federationapi/routing/routing.go

```diff
@@ -479,14 +479,14 @@ func Setup(
 	).Methods(http.MethodGet, http.MethodPost)
 
 	v1fedmux.Handle("/user/keys/claim", MakeFedAPI(
-		"federation_keys_claim", cfg.Matrix.ServerName, keys, wakeup,
+		"federation_keys_claim", cfg.Matrix.ServerName, cfg.Matrix.IsLocalServerName, keys, wakeup,
 		func(httpReq *http.Request, request *gomatrixserverlib.FederationRequest, vars map[string]string) util.JSONResponse {
 			return ClaimOneTimeKeys(httpReq, request, keyAPI, cfg.Matrix.ServerName)
 		},
 	)).Methods(http.MethodPost)
 
 	v1fedmux.Handle("/user/keys/query", MakeFedAPI(
-		"federation_keys_query", cfg.Matrix.ServerName, keys, wakeup,
+		"federation_keys_query", cfg.Matrix.ServerName, cfg.Matrix.IsLocalServerName, keys, wakeup,
```

Comment:
How does this work given we are passing in a single server name for the destination?

Reply:
We pass in both the *default* server name and the check function into `MakeFedAPI`. If the `X-Matrix` header contains a `destination=` field then we will match it using `cfg.Matrix.IsLocalServerName`. If there is no `destination=` then we use the given server name as the default value.